Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -197,35 +241,137 @@ namespace ROS2::Utils
+
+        const AZ::IO::Path importDirectoryTmp = AZ::IO::Path(AZ::Utils::GetProjectPath()) / "Assets" / "UrdfImporter" / directoryNameTmp;
+        const AZ::IO::Path importDirectoryDst = AZ::IO::Path(AZ::Utils::GetProjectPath()) / "Assets" / "UrdfImporter" / directoryNameDst;
+
+        // remove, if old import exists
+        fileIO->DestroyPath(importDirectoryTmp.c_str());
+        fileIO->DestroyPath(importDirectoryDst.c_str());
+
+        AZStd::string amentPrefixPath {enviromentalVariable};
+        AZStd::set<AZStd::string> files;
+
+        for (const auto& meshesFilename : meshesFilenames)
+        {
+
+            auto resolved = Utils::ResolveURDFPath(meshesFilename, urdFilename, amentPrefixPath);
+            if (!resolved.empty())
+            {
+                AZ::IO::Path resolvedPath (resolved);
+                AZ::IO::Path targetPathTmp (importDirectoryTmp/resolvedPath.Filename());
+                AZ::IO::Path targetPathDst (importDirectoryDst/resolvedPath.Filename());
+
+                const auto outcome = fileIO->Copy(resolvedPath.c_str(), targetPathTmp.c_str());
+                AZ_Printf("CopyAssetForURDF", "Copy %s to %s, result: %d", resolvedPath.c_str(), targetPathTmp.c_str(), outcome.GetResultCode());
+                if (outcome)
+                {
+                    copiedFiles.insert(targetPathDst.String());
+                    bool needsVisual = visuals.contains(meshesFilename);
+                    bool needsCollider = colliders.contains(meshesFilename);
+                    if (needsCollider||needsVisual){
+                        createSceneManifest(targetPathTmp.String(), needsCollider, needsVisual);
+                    }
+                }
+            }
+        }
+        // rename
+        const auto outcome = fileIO->Rename(importDirectoryTmp.c_str(), importDirectoryDst.c_str());
+        AZ_Printf("CopyAssetForURDF", "Rename %s to %s, result: %d", importDirectoryTmp.c_str(), importDirectoryDst.c_str(), outcome.GetResultCode());
+        if (!outcome)
+        {
+            copiedFiles.clear();
+        }
+        for (auto & copied : copiedFiles)
+        {
+            AZ_Printf("CopyAssetForURDF", " -> %s", copied.c_str());
+        }
+        return copiedFiles;
+    }
+
+    UrdfAssetMap FindAssetsForUrdf(const AZStd::unordered_set<AZStd::string>& meshesFilenames, const AZStd::string& urdFilename,
+                                   const AZStd::unordered_set<AZStd::string>& tip)
     {
         auto enviromentalVariable = std::getenv("AMENT_PREFIX_PATH");
         AZ_Error("UrdfAssetMap", enviromentalVariable, "AMENT_PREFIX_PATH is not found.");
         AZStd::string amentPrefixPath {enviromentalVariable};
 
         UrdfAssetMap urdfToAsset;
-        for (const auto& t : meshesFilenames)
+        // compute checksum for files from Tip
+        AZStd::unordered_map<AZ::Crc32,AZStd::string> tipFilesWithCrc;
```

Comment:
It might not be necessary to use the DB or CRCs for this.

When a source file appears on disk, GetSourceInfoBySourcePath should be available to ask for its UUID (subid zero)
once you have its UUID, GetAssetsProducedBySourceUUID can be used to enumerate all the product files generated by that particular source path, which gives you their relative path as well as their full AssetID (subid non zero).

It all happens asynchronously though, so you'd have to wait until GetAssetJobsInfo (for that source path) indicates that all jobs are done.   GetAssetJobsInfo is good to call anyway since it can optioanlly escalate the jobs it returns to the top of hte list in case there is a job queue.

GetAssetJobsInfo will return a vector of jobs that were created for that asset and each of those will have a status JobStatus m_status, so you can poll for when its complete.

This would allow you to get a definite list of all the output files for any given input file (DAE, STL, etc) as well as escalate those jobs to the top of the queue in case there is a queue as well as know if 1 failed.  The same API also lets you retrieve the build logs for the files which could be parsed to show any errors or warnings.

Reply:
You are right, it is no need, keep legacy mechanism for now. We can remove next PR.